Tests/Document/JSONObjectTests.swift

```diff
@@ -19,129 +19,109 @@ import XCTest
+                           {"boolean":"true","compB":{"compC":{"compD":{"id":"d-1"},"id":"c"},"id":"b"},"double":1.2222222,"integer":111,"long":9999999,"string":"abc"}
                            """)
 
-            _ = try? root.remove(member: \DocComponentA.string)
-            _ = try? root.remove(member: \DocComponentA.integer)
-            _ = try? root.remove(member: \DocComponentA.compB)
+            root.remove(key: "string")
+            root.remove(key: "integer")
+            root.remove(key: "compB")
 
-            XCTAssertEqual(root.toSortedJSON(),
+            XCTAssertEqual(root.debugDescription,
                            """
                            {"boolean":"true","double":1.2222222,"long":9999999}
                            """)
         }
     }
-}
-
-class TestDocument<T: JSONSpec> {
-    func update(_ callback: (_ root: JSONObject<T>) -> Void) {
-        let root = JSONObject<T>(target: CRDTObject(createdAt: TimeTicket.initial), context: ChangeContext(id: ChangeID.initial, root: CRDTRoot()))
-        callback(root)
-    }
-}
-
-class DocComponentA: JSONSpec {
-    var compB: JSONObject<DocComponentB>?
-    var boolean: Bool?
-    var integer: Int32?
-    var long: Int64?
-    var double: Double?
-    var string: String?
-
-    required init() {}
-
-    static var keyMap: [AnyKeyPath: String] {
-        [
-            \DocComponentA.compB: "nested",
-            \DocComponentA.boolean: "boolean",
-            \DocComponentA.integer: "integer",
-            \DocComponentA.long: "long",
-            \DocComponentA.double: "double",
-            \DocComponentA.string: "string"
-        ]
-    }
-}
-
-class DocComponentB: JSONSpec {
-    var id: String?
-    var compC: JSONObject<DocComponentC>?
-
-    required init() {}
-
-    static var keyMap: [AnyKeyPath: String] {
-        [\DocComponentB.id: "id",
-         \DocComponentB.compC: "nested"]
-    }
-}
 
-class DocComponentC: JSONSpec {
-    var id: String?
-    var compD: JSONObject<DocComponentD>?
+    func test_can_set_with_dictionary() {
+        let target = TestDocument()
+        target.update { root in
+            root.set([
+                "boolean": true,
+                "integer": Int32(111),
+                "long": Int64(9_999_999),
+                "double": Double(1.2222222),
+                "string": "abc",
+                "compB": ["id": "b",
+                          "compC": ["id": "c",
+                                    "compD": ["id": "d-1"]]]
+            ])
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"boolean":"true","compB":{"compC":{"compD":{"id":"d-1"},"id":"c"},"id":"b"},"double":1.2222222,"integer":111,"long":9999999,"string":"abc"}
+                           """)
+            let compD = root.get(keyPath: "compB.compC.compD") as? JSONObject
```

Comment:
```get(keyPath:)``` takes a string that is concatenated with a dot and returns the result.
Please comment.

Reply:
Since the dot is a reserved character for the key, we can extend get(key:) to perform with keyPath.

eg. get(key: "string") or get(key: "compB.compC")